controllers/actions.github.com/ephemeralrunnerset_controller.go

```diff
@@ -122,6 +122,24 @@ func (r *EphemeralRunnerSetReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
+	// Create proxy secret if not present
+	if ephemeralRunnerSet.Spec.EphemeralRunnerSpec.Proxy != nil {
+		proxySecret := new(corev1.Secret)
+		if err := r.Get(ctx, types.NamespacedName{Namespace: ephemeralRunnerSet.Namespace, Name: proxyEphemeralRunnerSetSecretName(ephemeralRunnerSet)}, proxySecret); err != nil {
+			if !kerrors.IsNotFound(err) {
+				log.Error(err, "Unable to get ephemeralRunnerSet proxy secret", "namespace", ephemeralRunnerSet.Namespace, "name", proxyEphemeralRunnerSetSecretName(ephemeralRunnerSet))
+				return ctrl.Result{}, err
+			}
+
+			// Create a compiled secret for the runner pods in the runnerset namespace
+			log.Info("Creating a ephemeralRunnerSet proxy secret for the runner pods")
+			if err := r.createProxySecret(ctx, ephemeralRunnerSet, log); err != nil {
```

Comment:
should we make AutoScalingRunnerSet_controller to create/cleanup this secret?

the EphemeralRunnerSet will get recreated every time there is a pod template change.

Reply:
When talking about this with @nikola-jokic we decided to go this way for that very reason, even considering the extreme of creating one secret per runner but that was introducing way too overhead and duplication. We went this way to decouple the two controllers, otherwise the other extreme is using one secret for all (listener and runners): a lot easier to reason about but very coupled. I trust your and Nikola's opinions more, so up to you really!

Reply:
I am fine with the current approach. 👍